alerting/src/main/kotlin/org/opensearch/alerting/AlertService.kt

```diff
@@ -213,12 +241,33 @@ class AlertService(
         workflowRunContext: WorkflowRunContext?
     ): Alert {
         val currentTime = Instant.now()
+        val alertState = if (workflowRunContext?.auditDelegateMonitorAlerts == true) {
+            Alert.State.AUDIT
+        } else {
+            Alert.State.ERROR
+        }
         return Alert(
             id = id, monitor = monitor, trigger = NoOpTrigger(), startTime = currentTime,
-            lastNotificationTime = currentTime, state = Alert.State.ERROR, errorMessage = alertError?.message,
-            schemaVersion = IndexUtils.alertIndexSchemaVersion,
-            workflowId = workflowRunContext?.workflowId ?: "",
-            executionId = executionId ?: ""
+            lastNotificationTime = currentTime, state = alertState, errorMessage = alertError?.message,
+            schemaVersion = IndexUtils.alertIndexSchemaVersion, executionId = executionId, workflowId = workflowRunContext?.workflowId ?: ""
+        )
+    }
+
+    fun composeChainedAlert(
+        ctx: ChainedAlertTriggerExecutionContext,
+        executionId: String,
+        workflow: Workflow,
+        associatedAlertIds: List<String>
+    ): Alert {
+        return Alert(
+            startTime = Instant.now(),
+            lastNotificationTime = Instant.now(),
+            state = Alert.State.ACTIVE,
```

Comment:
If the previous alert was active for the chained alert, do we ignore that and not extend the existing alert and add more info there like query level and bucket level monitors? This logic is more similar to doc level, which would make less sense for chained alerts.

Reply:
Will discuss this more with folks and fix this logic

Reply:
But the underlying motivation was that the chained alert can be created for that particular execution and be associated with delegate monitor alerts from that execution. will have to see if extending an alert fits this thinking.